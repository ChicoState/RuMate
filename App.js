import React from 'react';
import firebase from 'firebase';
import { StyleSheet } from 'react-native';
import { useScreens } from 'react-native-screens';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Icon from 'react-native-vector-icons/FontAwesome';
/* our imports */
import HomeScreen from './src/screens/HomeScreen';
import TasksScreen from './src/screens/TasksScreen';
import AuthScreen from './src/screens/AuthScreen';
import CreateTaskScreen from './src/screens/CreateTaskScreen';
import AddRoommateScreen from './src/screens/AddRoomateScreen';
import CalendarScreen from './src/screens/CalendarScreen';
import BillsScreen from './src/screens/BillsScreen';
import CreateBillScreen from './src/screens/CreateBillScreen';
import CreateRoommateGroupScreen from './src/screens/CreateRoommateGroupScreen';
import MessageScreen from './src/screens/MessagesScreen';
import CreateConversationScreen from './src/screens/CreateConversationScreen';
import ConversationScreen from './src/screens/ConversationScreen';
import InvitationsScreen from './src/screens/InvitationsScreen';
import AccountScreen from './src/screens/AccountScreen';
import PriorityScreen from './src/screens/PriorityScreen';
import ChangeDetailsScreen from './src/screens/ChangeDetailsScreen';

useScreens();
// ################################ //
// Remove to debug errors/warnings  //
   console.disableYellowBox = true; //
// ################################ //
const firebaseConfig = {
  apiKey: "AIzaSyA_ZWMR-MFG_ZHNK4_WuEeHoLP9vzsY_Vk",
  authDomain: "rumate-faaeb.firebaseapp.com",
  databaseURL: "https://rumate-faaeb.firebaseio.com",
  projectId: "rumate-faaeb",
  storageBucket: "",
  messagingSenderId: "484837108351",
  appId: "1:484837108351:web:53a875e9546d00aa2dad4a"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const tabNavigator = createBottomTabNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="home" size={20} />
      )
    },
  },
  Tasks: {
    screen: TasksScreen,
    navigationOptions: {
      tabBarLabel: "Tasks",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="tasks" size={20} />
      )
    },
  },
  Bills: {
    screen: BillsScreen,
    navigationOptions: {
      tabBarLabel: "Bills",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="dollar" size={20} />
      )
    },
  },
  Messages: {
    screen: MessageScreen,
    navigationOptions: {
      tabBarLabel: "Messages",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="comments" size={20} />
      )
    },
  }
},
{
  initialRouteName: 'Home',
});

const taskNavigator = createStackNavigator({
  CreateTask: CreateTaskScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const appStackNavigator = createStackNavigator({
  Priority: PriorityScreen,
  Invitations: InvitationsScreen,
  CreateRoommateGroup: CreateRoommateGroupScreen,
  AddRoommate: AddRoommateScreen,
  Home: tabNavigator,
  Tasks: taskNavigator,
  CreateBill: CreateBillScreen,
  TaskCalendar: CalendarScreen,
  Messages: MessageScreen,
  CreateConversation: CreateConversationScreen,
  Conversation: ConversationScreen,
  Account: AccountScreen,
  ChangeDetails: ChangeDetailsScreen

},{
  initialRouteName: 'Home',
  defaultNavigationOptions: {
    header: null
  }
});

const authStack = createStackNavigator({
  Login: AuthScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const switchNav = createSwitchNavigator({
  App: appStackNavigator,
  Auth: authStack,
}, {
  initialRouteName: 'Auth',
  defaultNavigationOptions: {
    header: null
  }
});

const App = createAppContainer(switchNav);

const styles = StyleSheet.create({
  headerLeft: {
    fontSize: 20
  }
});

export default () => {
  return (
    <App />
  );
}
