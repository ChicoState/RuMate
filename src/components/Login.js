import React, { useState } from 'react';
import { 
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  StatusBar,
  KeyboardAvoidingView
} from 'react-native';
import firebase from 'firebase';
// import { Header } from 'react-native-elements';
import TitleBlock from './TitleBlock';
import AuthSpinner from './AuthSpinner';

const Login = ({
  navigation,
  email, setEmail,
  password, setPassword,
  register, setRegister, }) => {

  const [waiting, setWaiting] = useState(false);

  const authenticateUser = async () => {
    setWaiting(true)
    await firebase.auth().signInWithEmailAndPassword(email, password).catch((error) => {
      console.log(error.code, '\n', error.message);
    }).then((myError) => {
      setWaiting(false)
      if (myError) {
        navigation.navigate('Home');
      }
    });
  }

  if (!waiting) {
    return (
      <KeyboardAvoidingView 
        style={styles.background}
        behavior="padding">
          <StatusBar barStyle="light-content"/>
          <TitleBlock 
            title="RuMate" 
            color="white"
            fontSize={40}
            paddingTop={"40%"}
            paddingBottom={"20%"}
          />
          <TextInput style={styles.input}
            value={email}
            onChangeText={setEmail}
            placeholder="e-mail"
            placeholderTextColor= "#444"
            autoFocus={false}
          />
          <TextInput style={styles.input}
            value={password}
            onChangeText={setPassword}
            placeholder="password"
            placeholderTextColor= "#444"
          />
        <TouchableOpacity style={styles.submit} 
          onPress = {authenticateUser}>
          <Text style={[styles.lightText, styles.button]}>
            Login
          </Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.submit}
          onPress={() => {
            setRegister(!register);
          }} >
          <Text style={[styles.lightText, styles.button]}>
            Register
          </Text>
        </TouchableOpacity>
      </KeyboardAvoidingView>
    );
  } else {
    return <AuthSpinner />
  }
}

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#222',
    marginVertical: 5,
    width: 300,
    marginHorizontal: 5,
    alignSelf: 'center',
    color: 'white',
    fontSize: 18,
    paddingVertical: 10,
    paddingLeft: 5,
    backgroundColor: 'black'
  },
  lightText: {
    color: 'white'
  },
  button: {
    fontSize: 20,
    paddingTop: 10,
  },
  submit: {
    borderWidth: 2,
    borderColor: '#222',
    width: 200,
    alignItems: 'center',
    height: 50,
    marginHorizontal: 5,
    alignSelf: 'center',
    marginVertical: 5
  },
  background: {
    flex: 1,
    backgroundColor: '#111111'
  },
});

export default Login