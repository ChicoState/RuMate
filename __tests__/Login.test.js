import React from 'react';
import renderer from 'react-test-renderer';
import firebase from 'firebase'
import App from '../App'
import Login from '../src/components/Login';

describe('<Login />', () => {
  <App />
  it('has 2 child', () => {
    const tree = renderer.create(<Login />).toJSON()
    console.log(tree.children.length)
    expect(tree.children.length).toBe(2)
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});