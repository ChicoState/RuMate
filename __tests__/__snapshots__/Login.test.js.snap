// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(25,25,25)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "color": "white",
              "fontSize": 50,
              "paddingBottom": "20%",
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": "40%",
            },
          ]
        }
      >
        RuMate
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      placeholder="e-mail"
      placeholderTextColor="#444"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderColor": "#222",
          "borderRadius": 15,
          "borderWidth": 1,
          "color": "white",
          "fontSize": 18,
          "marginHorizontal": 5,
          "marginVertical": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 20,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      placeholder="password"
      placeholderTextColor="#444"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderColor": "#222",
          "borderRadius": 15,
          "borderWidth": 1,
          "color": "white",
          "fontSize": 18,
          "marginHorizontal": 5,
          "marginVertical": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 20,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "opacity": 0,
        "paddingTop": 20,
        "transform": Array [
          Object {
            "translateY": 100,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "green",
          "borderColor": "#222",
          "borderRadius": 15,
          "height": 50,
          "marginHorizontal": 5,
          "marginVertical": 5,
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
            },
            Object {
              "fontSize": 25,
              "paddingTop": 10,
            },
          ]
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "black",
          "borderColor": "#222",
          "borderRadius": 15,
          "height": 50,
          "marginHorizontal": 5,
          "marginVertical": 5,
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            Object {
              "fontSize": 25,
              "paddingTop": 10,
            },
          ]
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
